src/main/java/htsjdk/variant/vcf/VCFMetaDataLines.java

```diff
@@ -0,0 +1,298 @@
+package htsjdk.variant.vcf;
+
+import htsjdk.samtools.util.Log;
+import htsjdk.tribble.TribbleException;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+//TODO: should this class be public so consumers can use in place of Set<VCFHeaderLine>
+
+/**
+ * Class for managing a set of VCFHeaderLines for a VCFHeader.
+ */
+class VCFMetaDataLines implements Serializable {
+    public static final long serialVersionUID = 1L;
+    protected final static Log logger = Log.getInstance(VCFMetaDataLines.class);
+
+    // Require a unique key for each ID line
+    // Allow duplicate non-ID keys, unless fileformat, reference ?
+
+    // TODO: Should we reject attempts to add two contig header lines with the same contigIndex ?
+    // TODO: GATK VcfUtilsUnitTest.createHeaderLines test creates headers with contig lines with identical (0) indices
```

Comment:
Again, I think that it is not a good idea to block htsjdk development for failing tests in downstream projects...